src/Rules/UselessConstructs/NoUselessWithFunctionCallsRule.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+namespace NunoMaduro\Larastan\Rules\UselessConstructs;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\FuncCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Rules\Rule;
+
+/**
+ * @implements Rule<FuncCall>
+ */
+class NoUselessWithFunctionCallsRule implements Rule
+{
+    public function getNodeType(): string
+    {
+        return FuncCall::class;
+    }
+
+    public function processNode(Node $node, Scope $scope): array
+    {
+        /** @var FuncCall $node */
+        if (! $node->name instanceof Node\Name) {
+            return [];
+        }
+
+        if ($node->name->toString() !== 'with') {
+            return [];
+        }
+
+        $args = $node->getArgs();
+        if (count($args) === 1) {
+            return ["Calling the helper function 'with()' with only one argument simply returns the value itself. if you want to chain methods on a construct, use '(new ClassName())->foo()' instead"];
+        }
+
+        if ($args[1]->value instanceof Node\Expr\Closure || ($args[1]->value instanceof Node\Scalar\String_ && function_exists($args[1]->value->value))) {
```

Comment:
`$args` is not guaranteed to have index `1` here. Please do a check before.

Also, same as the previous comment; please use the type system to do the checks here.